src/server/game/Conditions/ConditionMgr.cpp

```diff
@@ -284,7 +284,11 @@ bool Condition::Meets(ConditionSourceInfo& sourceInfo) const
         }
         case CONDITION_NEAR_CREATURE:
         {
-            condMeets = GetClosestCreatureWithEntry(object, ConditionValue1, (float)ConditionValue2, bool(!ConditionValue3)) ? true : false;
+            Creature* crea = GetClosestCreatureWithEntry(object, ConditionValue1, (float)ConditionValue2, bool(!ConditionValue3));
+            if (!crea || (ConditionValue3 && crea->getDeathState() == DEAD))
```

Comment:
@Treeston I'm not sure i'm following, if ConditionValue3 is true, GetClosestCreatureWithEntry() will already have excluded all creatures with ALIVE deathState

@jackpoz I don't think change GetClosestCreatureWithEntry() to ignore despawned creatures is wise, it could be used like, for example, to find and respawn a dead creature. Also, I don't know how to create a grid searcher to do this :/